lib/temporal/client/implementation.rb

```diff
@@ -154,8 +154,7 @@ def handle_start_workflow(input)
       rescue Temporal::Bridge::Error => e
         # TODO: Raise a better error from the bridge
         if e.message.include?('AlreadyExists')
-          # TODO: Replace with a more specific error
-          raise Temporal::Error, 'Workflow already exists'
+          raise Temporal::Error::RPCError, 'Workflow already exists'
```

Comment:
Why not a `Temporal::Error::WorkflowExecutionAlreadyStarted` exception, similar to what you did for queries? IMHO it is not uncommon for users to catch that specific exception...

Reply:
We should do this in Python as well. Not only is there a benefit of catching this specific exception, but we need to extract https://github.com/temporalio/api/blob/9645a99d585b426ebb7555b8f61f5b1ed1aa42af/temporal/api/errordetails/v1/message.proto#L47 out of the status details and make the run ID visible to users.

Reply:
> we need to extract `WorkflowExecutionAlreadyStartedFailure` out of the status details and make the run ID visible to users.

Oh really?!? Is there such a thing?!? Never heard of those fields. Will have to implement in TypeScript too. Seems to be properly implemented in Java.

Reply:
Agree. Added it here — https://github.com/temporalio/sdk-ruby/pull/69/commits/debc542ed2d065038f41622649492dc5e74066de. I'll add the `run_id` to that exception separately (since it requires encoding/decoding the error raised by Rust). Here's issue to keep a track of it — #72